### PR TITLE
Completes relatedness and ancestry processes

### PR DESCRIPTION
**Pull request by aldembert**
This PR includes all the code cleaning, bug fixing and setup done for including Daniels new code updates from last week for running the full relatedness and ancestry components of the siteQC pipeline done by Vlad.

To test the pipeline in AWS do:

*install all standard dependencies.

git clone https://github.com/lifebit-ai/relate.git

cd relate

git checkout salvador-add-ancestry-mvp

docker build -t lifebitai/relate:2.0dev . -f Dockerfile

*Run aws configure and add the needed s3 configuration

*Run the nextflow pipeline with the required test files:

nextflow main.nf --inputDir 's3://lifebit-featured-datasets/projects/gel/siteqc/ARtestFiles/Annotation_newtest/' --input s3://lifebit-featured-datasets/projects/gel/siteqc/input.csv --inputMichiganLDfile s3://lifebit-featured-datasets/projects/gel/siteqc/MichiganLD_liftover_exclude_regions_PARSED.txt --inputMichiganLDfileExclude s3://lifebit-featured-datasets/projects/gel/siteqc/MichiganLD_liftover_exclude_regions_PARSED.txt --inputAncestryAssignmentProbs s3://lifebit-featured-datasets/projects/gel/siteqc/aggV2_R9_M30K_1KGP3_ancestry_assignment_probs.tsv --inputPCsancestryrelated s3://lifebit-featured-datasets/projects/gel/siteqc/aggV2_bedmerge_30KSNPs_labkeyV9_08062020_update_PCsancestryrelated.tsv --inputFinalPlatekeys s3://lifebit-featured-datasets/projects/gel/siteqc/sampleList.txt --inputProbs200K s3://lifebit-featured-datasets/projects/gel/siteqc/aggV2_ancestry_assignment_probs_1KGP3_200K.tsv --inputUNRELATED_1KGP3 s3://lifebit-featured-datasets/projects/gel/siteqc/UNRELATED_1KGP3.samples --input1KGP3 s3://lifebit-featured-datasets/projects/gel/siteqc/1KGP3.sample_table --inputSuper_pop_codes s3://lifebit-featured-datasets/projects/gel/siteqc/super_pop_codes.tsv --input05both1K100K_eigenvec s3://lifebit-featured-datasets/projects/gel/siteqc/1KGP3_30K_unrel_autosomes.eigenvec --inputGELprojection_proj_eigenvec s3://lifebit-featured-datasets/projects/gel/siteqc/1KGP3_30K_unrel_autosomes.eigenvec_TMPPROJ

This PR uses the docker container "lifebitai/relate:2.0dev"

The successfully completed job in cloudOS is at:

https://cloudos.lifebit.ai/app/jobs/5f6a1b0b3325bf011218c739

image

-------------------------------

Many thanks to contributing to lifebit-ai/relate!

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [x] `README.md` is updated

**Learn more about contributing:** https://github.com/lifebit-ai/relate/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -25,6 +25,8 @@ RUN R -e "install.packages('https://cran.r-project.org/src/contrib/Archive/BH/BH
 RUN R -e "install.packages('https://cran.r-project.org/src/contrib/Archive/dplyr/dplyr_0.8.5.tar.gz', repos=NULL, type='source')"
 RUN R -e "install.packages('https://cran.r-project.org/src/contrib/Archive/data.table/data.table_1.12.8.tar.gz', repos=NULL, type='source')"
 RUN R -e "install.packages('https://cran.r-project.org/src/contrib/magrittr_1.5.tar.gz', repos=NULL, type='source')"
+RUN R -e "withr::with_makevars(c(PKG_LIBS = '-liconv'), install.packages('tidyverse',repos='http://cran.us.r-project.org'), assignment = '+=')"
```
Comment (cgpu) [inline Dockerfile:28]: @aldembert out of curiosity, why are we running this with `withr::with_makevars`?

**Review by cgpu — CHANGES_REQUESTED** [review]
Thanks for the contribs @aldembert!

Do you mind also adding documentation on the new parameters you might have added?

I have tagged you where @Vlad-Dembrovskyi has added docs for `lifebit-ai/siteqc` here:
https://github.com/lifebit-ai/siteqc/pull/12/files#r493310896

The file you need to change is `docs/usage.md`. Have a look at the format
https://github.com/lifebit-ai/siteqc/blob/master/docs/usage.md#required-siteqc-pipeline-arguments

File: Dockerfile (hunk 1 above)
Comment (aldembert) [reply Dockerfile:28]: Very good question. I actually struggled a lot finding that solution. The problem is that without that specific way of running the installation of tidyverse library it wont recognize the required dependencies and environment variables that Tidyverse requires. Without that way of running the package installation this is what you get:

image
image
image


And after using the "withr" way of installing all gets done correctly for tidyverse:

image
image

Thats the reason for this @cgpu

Comment (cgpu) [reply Dockerfile:28]: Nice thanks for the explanation @aldembert 👍 !

**Review by cgpu — APPROVED** [review]
Goog to go 👍

**Pull request merged**